packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,518 @@
+
+  /// The slot placed on the end side of the screen; meaning on the right
+  /// when the textDirection is LTR and on the left when it is RTL.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? secondaryNavigation;
+
+  /// The slot placed on the top part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? topNavigation;
+
+  /// The slot placed on the bottom part of the screen.
+  ///
+  /// Note: if using flexibly sized Widgets like [Container], wrap the Widget in a
+  /// [SizedBox] or limit its size by another method.
+  final SlotLayout? bottomNavigation;
+
+  /// The slot that fills the rest of the space in the center.
+  final SlotLayout? body;
+
+  /// A supporting slot for [body]. Has a sliding entrance animation by default.
+  /// The default ratio for the split between [body] and [secondaryBody] is so
+  /// that the split axis is in the center of the screen.
+  final SlotLayout? secondaryBody;
+
+  /// Defines the fractional ratio of [body] to the [secondaryBody].
+  ///
+  /// For example 0.3 would mean [body] takes up 30% of the available space
+  /// and[secondaryBody] takes up the rest.
+  ///
+  /// If this value is null, the ratio is defined so that the split axis is in
+  /// the center of the screen.
+  final double? bodyRatio;
+
+  /// Whether or not the developer wants the smooth entering slide transition on
+  /// [secondaryBody].
+  ///
+  /// Defaults to true.
+  final bool internalAnimations;
+
+  /// The orientation of the body and secondaryBody. Either horizontal (side by
+  /// side) or vertical (top to bottom).
+  ///
+  /// Defaults to Axis.horizontal.
+  final Axis bodyOrientation;
+
+  @override
+  State<AdaptiveLayout> createState() => _AdaptiveLayoutState();
+}
+
+class _AdaptiveLayoutState extends State<AdaptiveLayout>
+    with TickerProviderStateMixin {
+  late AnimationController _controller;
+
+  late Map<String, SlotLayoutConfig?> chosenWidgets =
+      <String, SlotLayoutConfig?>{};
+  Map<String, Size?> slotSizes = <String, Size?>{};
+
+  Map<String, ValueNotifier<Key?>> notifiers = <String, ValueNotifier<Key?>>{};
+
+  Set<String> isAnimating = <String>{};
+
+  @override
+  void initState() {
+    if (widget.internalAnimations) {
+      _controller = AnimationController(
+        duration: const Duration(seconds: 1),
+        vsync: this,
+      )..forward();
+    } else {
+      _controller = AnimationController(
+        duration: Duration.zero,
+        vsync: this,
+      );
+    }
+
+    for (final _SlotIds item in _SlotIds.values) {
+      notifiers[item.name] = ValueNotifier<Key?>(null);
+      notifiers[item.name]!.addListener(() {
+        isAnimating.add(item.name);
+        _controller.reset();
+        _controller.forward();
+      });
```

Comment:
You can also write this like this:

```suggestion
      notifiers[item.name] = ValueNotifier<Key?>(null)
        ..addListener(() {
          isAnimating.add(item.name);
          _controller.reset();
          _controller.forward();
        });
```